File: SimpleRPG/SimpleRPG/Player.cs
﻿namespace SimpleRPG
{
    public sealed class Player : Entity
    {
        public int Experience { get; set; }
        public Player(string name = "Zero", int attack = 5, int defense = 5, int life = 100, int mana = 100, int stamina = 100,
            int experience = 0)
        {
            Name = name;
            Attack = attack;
            Defense = defense;
            Life = life;
            Mana = mana;
            Stamina = stamina;
            Experience = experience;
        }
    }
}

File: SimpleRPG/SimpleRPGMVC/wwwroot/js/player.ts
class Player extends Entity {
    private Experience = 0;

    constructor() {
        super();
        this.Name = "Unknown Player";
        this.Attack = 5;
        this.Defense = 5;
        this.Life = 100;
        this.Mana = 100;
        this.Stamina = 100;
    }

    get experience(): number {
        return this.Experience;
    }
}